mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java

```diff
@@ -70,15 +78,18 @@ public RabbitMQEventBus(RabbitMQConnectionFactory rabbitMQConnectionFactory, Eve
         this.retryBackoff = retryBackoff;
         this.eventDeadLetters = eventDeadLetters;
         this.isRunning = new AtomicBoolean(false);
+        this.sendOptions = new SendOptions()
+            .channelMono(connectionMono.map(Throwing.function(connection -> connection.createChannel())).cache())
```

Comment:
What happens in case of temporary rabbitMQ failure?

```
rabbitMQ.pause();

eventBus.dispatch(EVENT, xxx).block(); // will fail

rabbit.unpause();

eventBus.dispatch(EVENT, xxx).block(); // should succeed
```

I'm not sure we do support this know but this is super important... We should test this....

Reply:
Spoiler alert: our code fails at this test ^^

Reply:
using `cache()` might be the reason

Reply:
if `connectionMono` is also cached, then we have to re-create both, connection and channel :)